lib.php

```diff
@@ -257,31 +269,37 @@ function subcourse_scale_used_anywhere($scaleid) {
  */
 function mod_subcourse_cm_info_view(cm_info $cm) {
     global $CFG, $USER, $DB;
-    require_once($CFG->libdir.'/gradelib.php');
+    $config = get_config('subcourse');
+    $displayoptions = $DB->get_field('subcourse', 'displayoptions', ['id' => $cm->instance]);
+    $displayoptions = empty($displayoptions) ? array() : unserialize($displayoptions);
 
     $html = '';
 
-    $sql = "SELECT r.*
-              FROM {course} r
-              JOIN {subcourse} s ON s.refcourse = r.id
-             WHERE s.id = :subcourseid";
-
-    $refcourse = $DB->get_record_sql($sql, ['subcourseid' => $cm->instance], IGNORE_MISSING);
-
-    if ($refcourse) {
-        $percentage = \core_completion\progress::get_course_progress_percentage($refcourse);
+    $printprogress = empty($displayoptions) ? $config->coursepageprintprogress : $displayoptions['coursepageprintprogress'];
+    if ($printprogress) {
+        $sql = "SELECT r.*
+                  FROM {course} r
+                  JOIN {subcourse} s ON s.refcourse = r.id
+                 WHERE s.id = :subcourseid";
+
+        $refcourse = $DB->get_record_sql($sql, ['subcourseid' => $cm->instance], IGNORE_MISSING);
+        $percentage = null;
+        if ($refcourse) {
+            $percentage = \core_completion\progress::get_course_progress_percentage($refcourse);
+        }
         if ($percentage !== null) {
             $percentage = floor($percentage);
             $html .= html_writer::tag('div', get_string('currentprogress', 'subcourse', $percentage),
                 ['class' => 'contentafterlink']);
         }
     }
 
-    $currentgrade = grade_get_grades($cm->course, 'mod', 'subcourse', $cm->instance, $USER->id);
-
-    if (!empty($currentgrade->items[0]->grades)) {
-        $currentgrade = reset($currentgrade->items[0]->grades);
-        if (isset($currentgrade->grade) and !($currentgrade->hidden)) {
+    $printgrade = empty($displayoptions) ? $config->coursepageprintgrade : $displayoptions['coursepageprintgrade'];
```

Comment:
Any particular reason for these extra checks? Was there a problem with these?

Reply:
Good example of complications from serialized data usage...As could not set default value for existing instance I added checks  to use general config if not defined in module. Will fix by implementing 2 dedicated DB fields

Reply:
Ah, yes. I see. Yeah. And there are others that appear later in time. E.g. unable to edit things easily in the database. Or, when a new option is added later but then you restore an old module that still has only two values serialized etc. I have never had good experience with this way of storing things. Plus, if we went this way, i would suggest to use JSON instead of PHP-native one.